server/cli/init/base-file-sources/world.js

```diff
@@ -23,20 +24,46 @@ module.exports = function () {
     };
 
     /* eslint-disable new-cap */
-    this.Before(function (callback) {
+    this.Before(function (scenario, callback) {
     /* eslint-enable new-cap */
         global.httpBackend = new HttpBackend(global.browser);
         callback();
     });
 
     /* eslint-disable new-cap */
-    this.After(function (callback) {
+    this.After(function (scenario, callback) {
+        var browserlogArray = [];
+        var message;
     /* eslint-enable new-cap */
         global.httpBackend.clear();
-        global.browser.manage.deleteAllCookies();
+        global.browser.manage().deleteAllCookies();
         global.browser.executeScript('window.sessionStorage.clear();');
         global.browser.executeScript('window.localStorage.clear();');
-        callback();
+        if (scenario.isFailed()) {
+            global.browser.takeScreenshot().then(function (base64png) {
+                var decodedImage = new Buffer(base64png, 'base64').toString('binary');
+                scenario.attach(decodedImage, 'image/png', callback);
+            }, function (err) {
+                callback(err);
+            })
+            global.browser.manage().logs().get('browser').then(function (browserlog) {
+                browserlog.forEach(function (log) {

```

Comment:
maybe this could be done with a filter and a map:

```javascript
var severeErrors = browserLog.filter(function (log) { 
    return log.level.name === SEVERE;
})
.map(function (log) {
    var message = log.message;
    return message.substring(message.indexOf('Error'), message.indexOf('\n'));
});
```